planner/core/find_best_task.go

```diff
@@ -570,6 +553,50 @@ func (ds *DataSource) findBestTask(prop *property.PhysicalProperty) (t task, err
 	return
 }
 
+func (ds *DataSource) canConvertToPointGet(candidate *candidatePath) bool {
+	path := candidate.path
+
+	canConvertPointGet := (!ds.isPartition && len(path.Ranges) > 0) || (ds.isPartition && len(path.Ranges) == 1)
+	canConvertPointGet = canConvertPointGet && candidate.path.StoreType != kv.TiFlash
+	if !candidate.path.IsTablePath {
+		canConvertPointGet = canConvertPointGet &&
+			candidate.path.Index.Unique &&
+			!candidate.path.Index.HasPrefixIndex() &&
+			len(candidate.path.Ranges[0].LowVal) == len(candidate.path.Index.Columns)
+	}
+	if !canConvertPointGet {
+		return false
+	}
+	allRangeIsPoint := true
+	for _, ran := range path.Ranges {
+		if !ran.IsPoint(ds.ctx.GetSessionVars().StmtCtx) {
+			allRangeIsPoint = false
+			break
+		}
+	}
+	if !allRangeIsPoint {
+		return false
+	}
+
+	if PreparedPlanCacheEnabled() {
+		hasParamMarkerConst := false
+		for _, cond := range path.AccessConds {
+			hasParamMarkerConst = hasParamMarkerConst || expression.ParamConstInExpression(cond)
+		}
+		if !hasParamMarkerConst {
+			return true
+		}
+		if path.IsTablePath && (len(path.AccessConds) != -1 || path.AccessConds[0].(*expression.ScalarFunction).FuncName.L != ast.EQ) {
```

Comment:
It seems to me that `!= -1`  can cover the later cond?

Reply:
this PR seems to be dup with https://github.com/pingcap/tidb/pull/17537/commits/607f86ea1533d1cd243dd7d2e09be7546f649cb9.

Reply:
`x.Handle = kv.IntHandle(x.Path.Ranges[0].LowVal[0].GetInt64())` is implemented in cherry pick, but not seems to be implemented here. PTAL @winoros

Reply:
Yeah, there's no `Kv.IntHandle` in the released branch. So keep it this way instead.